sycl/include/CL/sycl/intel/sub_group.hpp

```diff
@@ -168,30 +180,45 @@ struct sub_group {
   /* --- collectives --- */
 
   template <typename T>
+#if __cplusplus >= 201402L
+  __SYCL_DEPRECATED__("Use sycl::intel::broadcast instead.")
+#endif
   EnableIfIsScalarArithmetic<T> broadcast(T x, id<1> local_id) const {
     return detail::spirv::GroupBroadcast<__spv::Scope::Subgroup>(x, local_id);
   }
 
   template <typename T, class BinaryOperation>
+#if __cplusplus >= 201402L
+  __SYCL_DEPRECATED__("Use sycl::intel::reduce instead.")
+#endif
   EnableIfIsScalarArithmetic<T> reduce(T x, BinaryOperation op) const {
     return detail::calc<T, __spv::GroupOperation::Reduce,
                         __spv::Scope::Subgroup>(
         typename detail::GroupOpTag<T>::type(), x, op);
   }
 
   template <typename T, class BinaryOperation>
+#if __cplusplus >= 201402L
+  __SYCL_DEPRECATED__("Use sycl::intel::reduce instead.")
+#endif
   EnableIfIsScalarArithmetic<T> reduce(T x, T init, BinaryOperation op) const {
     return op(init, reduce(x, op));
   }
 
   template <typename T, class BinaryOperation>
+#if __cplusplus >= 201402L
```

Comment:
You don't need this `#if` anymore.

Reply:
Other than this small clean-up, the patch looks good to me.

Reply:
Oh, you're right.  Thanks!